### DISCUSSION TITLE
Recommended FORTRAN debugger?

### DISCUSSION THREAD

**Discussion by nbelakovski** [libprima/prima#47] [General]
I'm using VSCode with some extensions to debug the FORTRAN code, but it's not great. I can't inspect arrays with more than one dimension, and sometimes even 1-dimensional arrays have issues with inspection. Any recommendations?

**Comment by zaikunzhang** [comment]
Hi @nbelakovski! 


First of all, I hope you are using Fortran rather than FORTRAN. They are quite different, and no one should use FORTRAN for any new project. 

What do you mean by debugging? Is it static code analysis or to find bugs in your code? For the latter, I use gdb. For the former, I just compile my code using ALL Fortran compilers available (there are more than 10) with the most restrictive flags (e.g, `-pedantic -Wall -Wextra -Werror` for gfortran) and make sure that my code passes without ANY warning. There exists commercial static analysis tools but I have never used them. You should also try Fortran Language Server, which is available under VSC. See https://github.com/fortran-lang/vscode-fortran-support.

**Reply by nbelakovski** [reply]
Understood re: FORTRAN vs Fortran, will make sure to use the latter going forward.

I meant for finding bugs in the code. `gdb` is pretty hardcore! I like having a nice visual environment for debugging. I managed to discover shortly after posting this that the reason I wasn't able to inspect certain variables in VSCode was that they had been created in the functions calling the function I was in. If I used the tools to select the appropriate function in the call stack I could inspect the variables. Mildly annoying to have to go up and back down but better to have the functionality in some fashion rather than not at all.
